main.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"net/url"
+)
+type (
+	Visitor struct {
+		Host       string
+		UserAgent  string
+		RequestUri string
+		Headers    http.Header
+		QueryParams url.Values
+	}
+)
+
+func main() {
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		jsonVisitorData, err := json.MarshalIndent(
+			Visitor{
+				r.Host,
```

Comment:
This kind of initialisation has more downfalls than benefits there, could you please specify field names as well